### PR TITLE
Use CLBlast instead of clBLAS

### PR DESCRIPTION
**Pull request by vbkaisetsu**
This branch uses CLBlast instead of clBLAS in OpenCL device.

It is mainly for reducing dependencies. clBLAS requires Boost that is a large library, so it's difficult to adapt to multiple operating systems and devices.
CLBlast does not require additional library, so it is portable.

CLBlast keeps performance in my experiments.

### CODE REVIEW

File: cmake/FindCLBlast.cmake [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# - Try to find CLBlast lib
```
Comment (odashi) [inline cmake/FindCLBlast.cmake]: Not enough documentation. Describe all possible options and license.

**Review by odashi — CHANGES_REQUESTED** [review]

**Comment by odashi** [comment]
The CLBlast prerequisite should be documented in all existing documents (we've also forgotten to add clBLAS prerequisites in current documents).

**Comment by vbkaisetsu** [comment]
I rewrote `FindCLBlast.cmake` based on `FindPrimitiv.cmake`, and fixed documentation.
Please review it.

File: docs/source/installation.rst [hunk 2]
```diff
@@ -146,3 +146,68 @@ searching failed or you want to switch them:
...
+
+
+Installing primitiv with OpenCL
+-------------------------------
+
+
+OpenCL backend requires OpenCL C++ headers (CLHPP) and CLBlast.
+
+
+Installing CLHPP
```
Comment (odashi) [inline docs/source/installation.rst]: Installing OpenCL C++ Headers

File: docs/source/installation.rst [hunk 3]
```diff
@@ -146,3 +146,68 @@ searching failed or you want to switch them:
     -DCUDA_TOOLKIT_ROOT_DIR=/path/to/cuda \
     -DCUDNN_ROOT_DIR=/path/to/cuda \
     -DPRIMITIV_USE_CUDA=ON
+
+
+Installing primitiv with OpenCL
+-------------------------------
+
+
+OpenCL backend requires OpenCL C++ headers (CLHPP) and CLBlast.
```
Comment (odashi) [inline docs/source/installation.rst]: Probably not necessary. Prerequisites are already documented in previous sections.

File: docs/source/installation.rst [hunk 4]
```diff
@@ -146,3 +146,68 @@ searching failed or you want to switch them:
...
+.. code-block:: shell
+
+  $ apt install wget
+  $ wget https://github.com/CNugteren/CLBlast/archive/1.2.0.tar.gz -O ./clblast.tar.gz
+  $ mkdir clblast
+  $ cd clblast
+  $ tar xf ../clblast.tar.gz --strip-components 1
+  $ mkdir build
+  $ cd build
+  $ cmake .. [OPTIONS]  # See: https://github.com/KhronosGroup/OpenCL-CLHPP
```
Comment (odashi) [inline docs/source/installation.rst]: Fix comment.

File: docs/source/installation.rst [hunk 5]
```diff
@@ -146,3 +146,68 @@ searching failed or you want to switch them:
...
+  $ make && make install
+
+
+Configuring primitiv with OpenCL
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+The following command configures to build OpenCL backend using
```
Comment (odashi) [inline docs/source/installation.rst]: OpenCL backend -> the OpenCL backend

**Comment by vbkaisetsu** [comment]
Fixed.

**Review by odashi — APPROVED** [review]

**Pull request merged**